java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture6/tasks/_3/UserService.java

```diff
@@ -1,11 +1,20 @@
 package com.zoolatech.lecture6.tasks._3;
 
-record UserService(UserRepository userRepository) {
+class UserService {
 
-    public String findUserEmail(String id) throws UserMissingException {
-        if (!userRepository.findUserEmail(id).isEmpty()) {
-            return userRepository.findUserEmail(id);
+    private final UserRepository userRepository;
+
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+    public String findUserEmail(String id) {
+        try {
+            if (!userRepository.findUserEmail(id).isEmpty()) {
+                return userRepository.findUserEmail(id);
+            }
+        } catch (UserMissingException userRepository) {
```

Comment:
`userRepository` - it's better to use a different name for the exception - currently, it hides an instance of UserRepository